packages/aws-cdk-lib/aws-rds/adr/aurora-serverless-v2.md

```diff
@@ -0,0 +1,233 @@
+provisioned instance as the writer along with a provisioned writer in tier 0-1
+(for failover) and then a serverless v2 reader in tier 2-15 (so it doesn't scale with the
+writer).
+
+If you are currently using auto scaling with provisioned instances, you should
+instead switch to instance scaling with serverless v2. This allows for a larger
+pool of serverless readers with the appropriate cluster capacity range. Vertical
+scaling at the serverless instance level is much faster compared to launching
+new instances.
+
+## Constraints
+
+Some of these are discussed in more detail above, but reposting here for
+clarity.
+
+1. Max `MaxCapacity` of 128 (256GB)
+2. Min `MinCapacity` of 0.5 (1GB) (*with some exceptions)
+3. Supports engine version MySQL 8+ & PostgreSQL 13+
+4. Scaling rate depends on capacity and promotion tier
+
+## Decision
+
+The major decision is whether we should update the existing
+`rds.DatabaseCluster` API to take into account all these new constraints or
+create an entirely new API (i.e. `rds.DatabaseClusterV2`). I am proposing that
+we update the existing API since this will allow existing users to migrate.
+
+The changes we would need to make is to switch the `instanceProps` property from
+required to optional and deprecate it. Add a new property `clusterInstances`
+which would look something like:
+
+_updates to core_
+```ts
+interface DatabaseClusterBaseProps {
+  ...
+  /**
+   * @deprecated - use clusterInstances instead
+   */
+  readonly instances?: number;
+  /**
+   * @deprecated - use clusterInstances instead
+   */
+  readonly instanceProps?: InstanceProps;
+  readonly clusterInstances?: ClusterInstances;
+}
+
+interface ClusterInstances {
+  readonly writer: IClusterInstance;
+  readonly readers?: IClusterInstance[];
+}
+
+class ClusterInstance implements IClusterInstance {
+  public static provisioned(id: string, props: ProvisionedClusterInstanceProps = {}): IClusterInstance {
+    return new ClusterInstance(id, {
+      ...props,
+      instanceType: ClusterInstanceType.provisioned(props.instanceType),
+    });
+  }
+
+  public static serverlessV2(id: string, props: ServerlessV2ClusterInstanceProps = {}): IClusterInstance {
+    return new ClusterInstance(id, {
+      ...props,
+      instanceType: ClusterInstanceType.serverlessV2(),
+    });
+  }
+  private constructor(private id: string, private readonly props: ClusterInstanceProps = {}) { }
+
+  public bind(scope: Construct, cluster: DatabaseCluster, props: ClusterInstanceBindOptions): IAuroraClusterInstance {
+    // new class to represent this concept
+    return new AuroraClusterInstance(scope, this.id, {
+      cluster,
+      ...this.props,
+      ...props,
+    });
+  }
+}
+```
+
+_user configuration_
+```ts
+new rds.DatabaseCluster(this, 'Cluster', {
+  engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_3_03_0 }),
+  clusterInstances: {
+    writer: ClusterInstance.provisioned('writer'),
```

Comment:
Surely `provisioned` needs more parameters. Are those elided for focus on this particular decision?

Reply:
Yes, i'll update the example to atleast show that there are properties here.